ReactNativeClient/lib/MdToHtml/webviewLib.js

```diff
@@ -4,7 +4,18 @@ webviewLib.handleInternalLink = function(event, anchorNode) {
 	const href = anchorNode.getAttribute('href');
 	if (href.indexOf('#') === 0) {
 		event.preventDefault();
+		let old_hash = location.hash;
+
 		location.hash = href;
+
+		// HACK 
+		// For some reason anchors at the bottom cause the webview to move itself
+		// so that the content is aligned with the top of the screen
+		// This basically refreshes the scroll view so that is returns to a normal
+		// position, the scroll positions stays correct though
+		// Additionally an anchor could not be clicked twice because the location
+		// would not change, this fixes that also
+		setTimeout(function() { location.hash = old_hash; }, 10);
```

Comment:
I've read the comment but I don't quite understand the change. Based on the code it means, first we jump to the requested anchor (`location.hash = href`) but then we moved back to the previous anchor (`location.hash = old_hash`)? I'm sure I'm missing something but could you clarify the logic here?

Reply:
That's exactly what's happening, this fixes 2 bugs present in the way webview handles anchors, the first is that when you click on an anchor, the url for the webview changes to reflect that movement, which has the unintended side effect of making the webview un-reactive to subsequent presses of the same anchor link. This fixes that issue by resetting the url after some tiny delay, so the webview will correctly scroll into place then reset to a standard url thus allowing the same link to be pressed again (and get the correct scrolling behavior).

The other bug this fixes is a bit weirder and is why I labelled this "hack". The webview tries to place anchors at the top of the page, which means if you have an anchor near the bottom of your page without much text below it, the webview scrolls to the bottom and would shrink to until it was able to fit the anchor at the top of the page. I suggest you remove the line I've added and give it  test for yourself, its very strange. What I found was that by resetting the url the webview would correctly scroll to the bottom but wouldn't shrink to display the anchor.
I don't believe this is absolutely the best fix (hence the hack warning) but I was pretty stuck on how else it could be fixed.